core/helpers/EEH_Line_Item.helper.php

```diff
@@ -1367,35 +1601,48 @@ public static function visualize(EE_Line_Item $line_item, $indentation = 0)
      * @return array        keys are line items for tickets IDs and values are their share of the running total,
-     *                                          plus the key 'total', and 'taxable' which also has keys of all the ticket IDs. Eg
-     *                                          array(
-     *                                          12 => 4.3
-     *                                          23 => 8.0
-     *                                          'total' => 16.6,
-     *                                          'taxable' => array(
-     *                                          12 => 10,
-     *                                          23 => 4
-     *                                          ).
-     *                                          So to find which registrations have which final price, we need to find which line item
-     *                                          is theirs, which can be done with
-     *                                          `EEM_Line_Item::instance()->get_line_item_for_registration( $registration );`
+     *                                                  plus the key 'total', and 'taxable' which also has keys of all
+     *                                                  the ticket IDs.
+     *                                                  Eg array(
+     *                                                      12 => 4.3
+     *                                                      23 => 8.0
+     *                                                      'total' => 16.6,
+     *                                                      'taxable' => array(
+     *                                                          12 => 10,
+     *                                                          23 => 4
+     *                                                      ).
+     *                                                  So to find which registrations have which final price, we need
+     *                                                  to find which line item is theirs, which can be done with
+     *                                                  `EEM_Line_Item::instance()->get_line_item_for_registration(
+     *                                                  $registration );`
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
      */
-    public static function calculate_reg_final_prices_per_line_item(EE_Line_Item $line_item, $billable_ticket_quantities = array())
-    {
+    public static function calculate_reg_final_prices_per_line_item(
+        EE_Line_Item $line_item,
+        $billable_ticket_quantities = array()
+    ) {
+        $running_totals = [];
         // init running grand total if not already
-        if (!isset($running_totals['total'])) {
+        if (! isset($running_totals['total'])) {
             $running_totals['total'] = 0;
```

Comment:
Not sure what the need for the `isset` checks is here.  What about instead of initializing an empty array, you do (and thus remove the `isset` checks as well:

```php
$running_totals = [
	'total' => 0,
	'taxable' => [ 'total' => 0 ]
];
```

Reply:
me likey! will change